tensorboard/program.py

```diff
@@ -352,33 +352,37 @@ def __init__(self, wsgi_app, flags):
 
     # base_port: what's the first port to which we should try to bind?
     # should_scan: if that fails, shall we try additional ports?
-    (base_port, should_scan) = (
-        (flags.port, False)
-        if flags.port is not None
-        else (core_plugin.DEFAULT_PORT, True)
-    )
+    # max_attempts: how many ports shall we try?
+    should_scan = flags.port is None
+    base_port = core_plugin.DEFAULT_PORT if flags.port is None else flags.port
+    max_attempts = 10 if should_scan else 1
+
     if base_port > 0xFFFF:
       raise TensorBoardServerException(
           'TensorBoard cannot bind to port %d > %d' % (base_port, 0xFFFF)
       )
     max_attempts = 10 if should_scan else 1
-    base_port = min(base_port + max_attempts, 65536) - max_attempts
+    base_port = min(base_port + max_attempts, 0x10000) - max_attempts
```

Comment:
Re-reading this logic, is this line really necessary?  It can't actually be reached in the current code, it would only happen if we set the default base_port to a number within 10 of 65536.

Reply:
That’s correct. I should add to the test plan that I did in fact test
this by changing the `core_plugin.DEFAULT_PORT` to 65534 and observing
that it binds to 65526 instead.

(Arguably it would be nicer to count _down_ in such a case, but for code
that is never executed I have a limited budget for aesthetics.)